### PR TITLE
LAPI code cleanup (pure refactoring)

### PR DESCRIPTION
**Pull request by nmarton-da**
removing dead code

changelog_begin
changelog_end

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description
- [ ] If you mean to change the status of a component, please make sure you keep [the Component Status page](https://github.com/digital-asset/daml/blob/main/docs/source/support/component-statuses.rst) up to date.

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

**Review by skisel-da — APPROVED** [review]

**Review by rgugliel-da — APPROVED** [review]
LGTM, thanks for checking !

File: ledger/participant-integration-api/src/main/scala/platform/apiserver/TimedIndexService.scala [hunk 1]
```diff
@@ -45,11 +44,6 @@ private[daml] final class TimedIndexService(delegate: IndexService, metrics: Met
   ): Future[Option[DamlLf.Archive]] =
     Timed.future(metrics.daml.services.index.getLfArchive, delegate.getLfArchive(packageId))
 
-  override def getLfPackage(packageId: Ref.PackageId)(implicit
-      loggingContext: LoggingContext
-  ): Future[Option[Ast.Package]] =
-    Timed.future(metrics.daml.services.index.getLfPackage, delegate.getLfPackage(packageId))
```
Comment (tudor-da) [inline ledger/participant-integration-api/src/main/scala/platform/apiserver/TimedIndexService.scala:51]: You can remove this metric

File: ledger/participant-integration-api/src/main/scala/platform/index/MeteredReadOnlyLedger.scala [hunk 2]
```diff
@@ -140,11 +139,6 @@ private[platform] class MeteredReadOnlyLedger(ledger: ReadOnlyLedger, metrics: M
   ): Future[Option[Archive]] =
     Timed.future(metrics.daml.index.getLfArchive, ledger.getLfArchive(packageId))
 
-  override def getLfPackage(packageId: Ref.PackageId)(implicit
-      loggingContext: LoggingContext
-  ): Future[Option[Ast.Package]] =
-    Timed.future(metrics.daml.index.getLfPackage, ledger.getLfPackage(packageId))
```
Comment (tudor-da) [inline ledger/participant-integration-api/src/main/scala/platform/index/MeteredReadOnlyLedger.scala:146]: You can remove this metric

**Review by tudor-da — APPROVED** [review]
lgtm provided you remove the two `getLfPackage` unused metrics

**Review by mziolekda — APPROVED** [review]

**Pull request merged**